jetty-plus/src/main/java/org/eclipse/jetty/plus/annotation/LifeCycleCallback.java

```diff
@@ -155,31 +153,33 @@ public Method findMethod (Package pack, Class<?> clazz, String methodName, boole
         }
     }
 
+    
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(_className, _methodName);
+    }
+
+
     @Override
     public boolean equals (Object o)
     {
         if (o==null)
             return false;
-        if (!(o instanceof LifeCycleCallback))
-            return false;
-        LifeCycleCallback callback = (LifeCycleCallback)o;
 
-        if (callback.getTargetClass()==null)
-        {
-            if (getTargetClass() != null)
-                return false;
-        }
-        else if(!callback.getTargetClass().equals(getTargetClass()))
-           return false;
-        if (callback.getTarget()==null)
-        {
-            if (getTarget() != null)
-                return false;
-        }
-        else if (!callback.getTarget().equals(getTarget()))
+        if (!LifeCycleCallback.class.isInstance(o))
             return false;
 
-        return true;
+        LifeCycleCallback callback = (LifeCycleCallback)o;
+        
+        if (this == callback)
+            return true;
```

Comment:
Move this reference equality before the check `LifeCycleCallback.class.isInstance(o)`, so it'll early return and save the `instanceof`.